libsledge/src/memory_instructions.c

```diff
@@ -67,50 +109,93 @@ INLINE void
+		fprintf(stderr, "set_f64 OOB: offset %u + length %u > size %lu\n", offset, (uint32_t)sizeof(double),
+		        sledge_abi__current_wasm_module_instance.memory.size);
+		sledge_abi__wasm_trap_raise(WASM_TRAP_OUT_OF_BOUNDS_LINEAR_MEMORY);
+	}
+
 	*(double *)&sledge_abi__current_wasm_module_instance.memory.buffer[offset] = value;
 }
 
 INLINE void
 set_i8(uint32_t offset, int8_t value)
 {
 	assert(sledge_abi__current_wasm_module_instance.memory.buffer != NULL);
-	assert((uint64_t)offset + sizeof(int8_t) <= sledge_abi__current_wasm_module_instance.memory.size);
+
+	if (unlikely((uint64_t)offset + (uint64_t)sizeof(int8_t)
+	             > sledge_abi__current_wasm_module_instance.memory.size)) {
+		fprintf(stderr, "set_i8 OOB: offset %u + length %u > size %lu\n", offset, (uint32_t)sizeof(int8_t),
+		        sledge_abi__current_wasm_module_instance.memory.size);
+		sledge_abi__wasm_trap_raise(WASM_TRAP_OUT_OF_BOUNDS_LINEAR_MEMORY);
+	}
+
 	*(int8_t *)&sledge_abi__current_wasm_module_instance.memory.buffer[offset] = value;
 }
 
 INLINE void
 set_i16(uint32_t offset, int16_t value)
 {
 	assert(sledge_abi__current_wasm_module_instance.memory.buffer != NULL);
-	assert((uint64_t)offset + sizeof(int16_t) <= sledge_abi__current_wasm_module_instance.memory.size);
+
+	if (unlikely((uint64_t)offset + (uint64_t)sizeof(int16_t)
+	             > sledge_abi__current_wasm_module_instance.memory.size)) {
+		fprintf(stderr, "set_i16 OOB: offset %u + length %u > size %lu\n", offset, (uint32_t)sizeof(int16_t),
+		        sledge_abi__current_wasm_module_instance.memory.size);
+		sledge_abi__wasm_trap_raise(WASM_TRAP_OUT_OF_BOUNDS_LINEAR_MEMORY);
+	}
+
 	*(int16_t *)&sledge_abi__current_wasm_module_instance.memory.buffer[offset] = value;
 }
 
 INLINE void
 set_i32(uint32_t offset, int32_t value)
 {
 	assert(sledge_abi__current_wasm_module_instance.memory.buffer != NULL);
-	assert((uint64_t)offset + sizeof(int32_t) <= sledge_abi__current_wasm_module_instance.memory.size);
+
+	if (unlikely((uint64_t)offset + (uint64_t)sizeof(int32_t)
+	             > sledge_abi__current_wasm_module_instance.memory.size)) {
+		fprintf(stderr, "set_i32 OOB: offset %u + length %u > size %lu\n", offset, (uint32_t)sizeof(int32_t),
+		        sledge_abi__current_wasm_module_instance.memory.size);
+		sledge_abi__wasm_trap_raise(WASM_TRAP_OUT_OF_BOUNDS_LINEAR_MEMORY);
+	}
+
 	*(int32_t *)&sledge_abi__current_wasm_module_instance.memory.buffer[offset] = value;
 }
 
 INLINE void
 set_i64(uint32_t offset, int64_t value)
 {
 	assert(sledge_abi__current_wasm_module_instance.memory.buffer != NULL);
-	assert((uint64_t)offset + sizeof(int64_t) <= sledge_abi__current_wasm_module_instance.memory.size);
+
+	if (unlikely((uint64_t)offset + (uint64_t)sizeof(int64_t)
+	             > sledge_abi__current_wasm_module_instance.memory.size)) {
+		fprintf(stderr, "set_i64 OOB: offset %u + length %u > size %lu\n", offset, (uint32_t)sizeof(int64_t),
```

Comment:
Maybe all of these can just be a call to a function to handle the error instead of multiple function calls with many arguments here?